### ISSUE TITLE
add CREEP_FILL extrapolation method, needs https://github.com/esmf-org/esmf/pull/18

### ISSUE THREAD

**Issue by raphaeldussin** [pangeo-data/xESMF#21]
once PR https://github.com/esmf-org/esmf/pull/18 is merged (and new ESMPy released?)
add the method to xESMF supported extrapolation methods

**Comment by huard** [comment]
A new conda build for ESMPy seems to be available.

**Comment by raphaeldussin** [comment]
yes, it's on my todo list, I want to test and document the several extrapolation methods

**Comment by raphaeldussin** [comment]
@huard I noticed it too! I need to work on documenting the extrapolation methods and make a PR to include the new method. my bandwidth is lower than a dial-up modem right now :(

**Comment by huard** [comment]
No pressure! I was just noting it for future reference.

**Comment by imcslatte** [comment]
Hello, 

I was wondering if CREEP fill extrapolation would be added to xesmf? It would be a really usefull tool.

Thanks, 

Eli

**Comment by alioacar** [comment]
Hi everyone :)
I've needed this kind of extrapolation (especially in an on-the-fly manner) for the DTO applications. Calculating the weights and reloading them in xESMF works really well when dealing with AI emulators across multiple grids, so I jumped in here.
Since xESMF is well structured, I implemented the `creep_fill` in a fairly straightforward way. Below I'm sharing a comparison using ROMS output that I found in the example notebooks: first xESMF bilinear, then native ESMPy bilinear + creep_fill extrapolation and finally xESMF bilinear + creep_fill. So basically, I created a coarse source grid and regrid it back onto a finer grid for all.
<img width="9000" height="1500" alt="Image" src="https://github.com/user-attachments/assets/a6dda6ec-77bb-44bf-a5db-32c4f4635d5b" />
It seems to be working quite well. I'm currently adding guards. There are two main ones:
- `extrap_num_levels` must be provided 
- reject `creep_fill` with conservative methods

For the remaining cases, I think we can rely on ESMF/ESMPy layer to raise the errors. I'll clean things up and plan to open a PR.
Best,
Ali

**Comment by huard** [comment]
Thanks for taking this on !
